typegate/tests/type_nodes/array_of_optional.py

```diff
@@ -0,0 +1,49 @@
+from typegraph import policies
+from typegraph import t
+from typegraph import TypeGraph
+from typegraph.runtimes.deno import PredefinedFunMat
+
+with TypeGraph("array_optional") as g:
+    nested = t.struct(
+        {
+            "a": t.string(),
+            "b": t.integer(),
+            "c": t.struct({"c1": t.string(), "inner": t.array(t.string().optional())}),
+        }
+    )
+
+    rec = t.struct(
+        {
+            "struct_array": t.array(nested.optional()).optional(),
```

Comment:
I think we could also add a test with `t.optional(t.array())`, I think it's handled a little different from `t.array().optional` so maybe is good to also test it.

Reply:
As far as I know, in both cases the optional node is visited first which means both generate the same schema.

Reply:
I see, you're right.